apollo-router-core/src/services/checkpoint.rs

```diff
@@ -0,0 +1,299 @@
+use futures::future::BoxFuture;
+use std::sync::Arc;
+use tower::{Layer, Service};
+
+/// An enum representing the next step
+/// A service should follow
+pub enum Step<Request, Response>
+where
+    Request: Send + 'static,
+    Response: Send + 'static,
+{
+    /// `CheckpointService` should
+    /// Forward the call to the next service
+    Forward(Request),
```

Comment:
Nit, would prefer `Continue`, not sure I can provide an adequate justification though.

Reply:
I think it's because the the service is called `checkpoint`.
You continue through a physical checkpoint.
A physical checkpoint would not forward you.

Reply:
fixed in ee986a9d